plugins/org.python.pydev/src/org/python/pydev/editor/correctionassist/heuristics/AssistFString.java

```diff
@@ -0,0 +1,105 @@
+package org.python.pydev.editor.correctionassist.heuristics;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.jface.text.BadLocationException;
+import org.eclipse.jface.text.IDocument;
+import org.eclipse.jface.text.ITypedRegion;
+import org.python.pydev.core.IPyEdit;
+import org.python.pydev.core.IPythonNature;
+import org.python.pydev.core.MisconfigurationException;
+import org.python.pydev.core.docutils.PySelection;
+import org.python.pydev.core.partition.PyPartitionScanner;
+import org.python.pydev.core.proposals.CompletionProposalFactory;
+import org.python.pydev.editor.PyEdit;
+import org.python.pydev.editor.codefolding.PySourceViewer;
+import org.python.pydev.editor.correctionassist.IAssistProps;
+import org.python.pydev.shared_core.code_completion.ICompletionProposalHandle;
+import org.python.pydev.shared_core.code_completion.IPyCompletionProposal;
+import org.python.pydev.shared_core.image.IImageCache;
+import org.python.pydev.shared_core.image.IImageHandle;
+import org.python.pydev.shared_core.image.UIConstants;
+import org.python.pydev.shared_core.partitioner.FastPartitioner;
+import org.python.pydev.shared_core.string.FastStringBuffer;
+
+public class AssistFString implements IAssistProps {
+    @Override
+    public List<ICompletionProposalHandle> getProps(PySelection ps, IImageCache imageCache, File f,
+            IPythonNature nature, IPyEdit edit, int offset) throws BadLocationException, MisconfigurationException {
+        PySourceViewer viewer = null;
+        if (edit instanceof PyEdit) { //only in tests it is actually null
+            viewer = ((PyEdit) edit).getPySourceViewer();
+        }
+        List<ICompletionProposalHandle> l = new ArrayList<ICompletionProposalHandle>();
```

Comment:
Minor: rename from `l` to `lst` (`l` isn't a good variable name because it's too close to `1`).